docs/developers-guide.md

```diff
@@ -1,8 +1,190 @@
+
+**In a single plugin.js definition file can be registered plugins to multiple extension points.**
+
+The above `plugin.js` definition can be extended like this: 
+```javascript
+PluginRegistry.add('route', {
+    'url': '/autocomplete',
+    'module': 'graphdb.framework.autocomplete',
+    'path': 'autocomplete/app',
+    'chunk': 'autocomplete',
+    'controller': 'AutocompleteCtrl',
+    'templateUrl': 'pages/autocomplete.html',
+    'title': 'Autocomplete index',
+    'helpInfo': 'The Autocomplete index is used for automatic completion of URIs in the SPARQL editor and the View resource page. Use this view to enable or disable the index and check its status.'
+});
+
+PluginRegistry.add('main.menu', {
+    'items': [
+        {label: 'Setup', href: '#', order: 5, role: 'IS_AUTHENTICATED_FULLY', icon: "icon-settings"},
+        {label: 'Autocomplete', href: 'autocomplete', order: 40, parent: 'Setup', role: "IS_AUTHENTICATED_FULLY"}
+    ]
+});
+```
+
+Defining an extension point:
+```javascript
+// src/app.js
+// 1. Get all registered extension for the "route" extension point. 
+let routes = PluginRegistry.get('route');
+// 2. Loop through all extensions
+routes.forEach(function (route) {
+    // 3. Register every extension with the $routeProvider
+    $routeProvider.when(route.url, {
+        controller: route.controller,
+        templateUrl: route.templateUrl,
+        title: route.title,
+        helpInfo: route.helpInfo,
+        reloadOnSearch: route.reloadOnSearch !== undefined ? route.reloadOnSearch : true
+    });
+});
+```
+Above is an example of extension point in the workbench which allows registering modules that can be
+accessed by accessing different routes.
+
+Plugins can have the following attributes: `order`, `priority`, `disabled`.
+* The plugin `order` is optional and can be used when the extension point contract requires plugins 
+to be loaded in particular order.
+```javascript
+// src/module_1/plugin.js
+PluginRegistry.add('extension.point', {
+    'name': 'plugin-1',
+    'order': 10
+});
+``` 
+```javascript
+// src/module_2/plugin.js
+PluginRegistry.add('extension.point', {
+    'name': 'plugin-2',
+    'order': 20
+});
+``` 
+```javascript
+// src/main.js
+const plugins = PluginRegistry.get('extension.point');
+// plugins[0] -> `plugin-1`
+// plugins[1] -> `plugin-2`
+```
+If order is not provided, then plugins are loaded in order of their registration which is the order 
+in which the plugin.js files are processed during the application bundling.
+
+* The `priority` attribute is optional and can be used when an ordered plugin should be overridden 
+by another plugin. If not provided, priority for every plugin is set by default to `0`. Having two 
+plugins with same order but different priority results in overriding the lower priority plugin by 
+the other. If there are two plugins with equal order and priority, then an error is thrown to warn 
+the developer for the issue.
```

Comment:
What if one of them is disabled? Will an error occur then?

Reply:
No, it won't.